scripts/resolve_i64/main.go

```diff
@@ -0,0 +1,159 @@
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	//"github.com/uber/zanzibar/codegen"
+	"go.uber.org/thriftrw/compile"
+	"html/template"
+	"bytes"
+	"io/ioutil"
+	"github.com/uber/zanzibar/runtime"
+	"runtime"
+	"path/filepath"
+	"strings"
+	"github.com/uber/zanzibar/codegen"
+	"github.com/pkg/errors"
+	"sort"
+)
+
+const (
+	templateFileName = "i64.tmpl"
+	outFileName = "/types_i64.go"
+)
+
+type naivePackageNameResolver struct {
+}
+
+func (r *naivePackageNameResolver) TypePackageName(
+	thriftFile string,
+) (string, error) {
+	if thriftFile[0] == '.' {
+		return "", errors.Errorf("Naive does not support relative imports")
+	}
+
+	return "", nil
+}
+
+// Meta is the struct container for i64 related meta data and package name
+type Meta struct {
+	PackageName string
+	Types []I64Struct
+}
+
+// I64Structs is the struct container for array if I64Struct
+type I64Structs []I64Struct
+
+// I64Struct is the struct container for i64 related meta data
+type I64Struct struct {
+	IsLong bool
+	IsTimestamp bool
+	TypedefType string
+}
+
+func (l I64Structs) Len() int           { return len(l) }
+func (l I64Structs) Less(i, j int) bool {
+	return l[i].IsTimestamp
+}
+func (l I64Structs) Swap(i, j int)      { l[i], l[j] = l[j], l[i] }
+
+func getDirName() string {
+	_, file, _, _ := runtime.Caller(0)
+	return zanzibar.GetDirnameFromRuntimeCaller(file)
+}
+
+func main() {
+	thriftFile := os.Args[1]
+	module, err := compile.Compile(thriftFile)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to parse thrift file: %s", thriftFile))
+	}
+	meta := &Meta{}
+
+	s := strings.TrimSuffix(thriftFile, ".thrift")
+	s = strings.Replace(s, "/idl/", "/build/gen-code/", 1)
+	meta.PackageName = filepath.Base(s)
+
+	for _, typeDef := range module.Types {
+		t, ok := typeDef.(*compile.TypedefSpec)
+		if !ok {
+			continue
+		}
+		i64Struct := I64Struct{}
+		if t.Target != nil {
+			typThriftAnnotation := t.Target.ThriftAnnotations()
+			if typThriftAnnotation != nil {
+				p := naivePackageNameResolver{}
+				refType, err := codegen.GoReferenceType(&p, t)
+				if err != nil {
+					fmt.Fprintln(os.Stderr, fmt.Errorf("error parsing reference type: %s", err.Error()))
+					os.Exit(1)
+					return
+				}
+
+				i64Struct.TypedefType = refType[1:]
+				if typThriftAnnotation["json.type"] == "Long" {
+					i64Struct.IsLong = true
+				}
+				if typThriftAnnotation["json.type"] == "Date" {
+					i64Struct.IsTimestamp = true
+				}
+			}
+		}
+		if i64Struct.IsTimestamp || i64Struct.IsLong {
+			meta.Types = append(meta.Types, i64Struct)
+		}
+	}
+	// Note type defs are not read by unique order so they need to be sorted before writing
+	sort.Sort(I64Structs(meta.Types))
+	tmpl, err := template.ParseFiles(filepath.Join(getDirName(), templateFileName))
+	if err != nil {
+		fmt.Fprintln(os.Stderr, fmt.Errorf(
+			"error parsing template %s: %s", templateFileName, err.Error()))
+		os.Exit(1)
+		return
+	}
+	tplBuffer := bytes.NewBuffer(nil)
+	err = tmpl.Execute(tplBuffer, meta)
```

Comment:
if `len(meta.Types) == 0` we can just end here and not do any templating at all.